src/Compilers/Core/Portable/RuleSet/RuleSet.cs

```diff
@@ -332,7 +332,7 @@ private static ReportDiagnostic GetDiagnosticOptionsFromRulesetFile(Dictionary<s
             }
             catch (IOException e)
             {
-                if (e is FileNotFoundException || e is DirectoryNotFoundException)
+                if (e is FileNotFoundException || e.GetType().Name == "DirectoryNotFoundException")
```

Comment:
I'm hearing misleading statements, by not caring about ".NET Native" here, you are discounting UAP usage. I've been told UAP support is cared about.